### DISCUSSION TITLE
Overwriting function get_alias_candidates for Entity Linking with Knowledge Base

### DISCUSSION THREAD

**Discussion by nkaramooz** [explosion/spaCy#11400] [Help: Coding & Implementations] [answered]
I'm trying to see if there is a way of currently overwriting the Knowledge Base get_alias_candidates function that is used for entity linking. The reason for doing so is to incorporate fuzzy matching in candidate generation. I came across the following page regarding this topic:

https://github.com/explosion/spaCy/pull/5876

But when using this format, I get an error that says "Unknown function registry: 'assets' ". I also tried overwriting the function directly, but this failed because the function is read-only. 
`kb.get_alias_candidates = get_custom_candidates`

I've spent a while trying to search other parts of the documentation but haven't been able to find anything outlining how to go about it. Any help would be appreciated.

**Comment by rmitsch** [comment, marked as answer]
Hi @nkaramooz! It's sufficient to modify the configuration for your entity linker so that the entry for `get_candidates` references the correct function. This can be done programatically (see example in the [`EntityLinker` docs](https://spacy.io/api/entitylinker#config)) or in your `.yaml` config (see this [NEL tutorial project](https://github.com/explosion/projects/blob/v3/tutorials/nel_emerson/configs/nel.cfg#L40)).

The error sounds like you're trying to use a non-existent function registry - try using `@spacy.registry.misc("WhateverYouWannaCallYourRegistryEntry.v1")` instead. In general, please always include a minimal reproducible example so that we can reproduce your issues.

**Comment by nkaramooz** [comment]
Thanks for your answer! This worked though has created some downstream problems, which I am working through (mainly wishing it were easier to make a candidate object from an entity_id), but should be able to solve. Thanks for your time!
